### ISSUE TITLE
one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [8, 512, 512, 1, 3]], which is output 0 of ExpandBackward, is at version 4; expected version 1 instead.

### ISSUE THREAD

**Issue by achao2013** [facebookresearch/pytorch3d#153]
render bacward error！
![image](https://user-images.githubusercontent.com/5026576/79348330-75076780-7f67-11ea-8e57-7b6fea4ee437.png)

------------------------------------------------------------------------------------------------
complement:
 rendereras[k] = MeshRenderer(
                        rasterizer=rasterizers[k],
                        shader=TexturedSoftPhongShader(
                            device=device, cameras=camas[k],lights=lights,materials=materials))

**Comment by gkioxari** [comment]
What you provide is non descriptive and we can't really parse this. You need to isolate the problem. The code you are running does not seem to be merely PyTorch3D code. It also consists of your own code. It is impossible for us to help you. Please isolate your code, break it into smaller pieces, provide the code snippets that cause the issue and report back. 

Note that we have [tests](https://github.com/facebookresearch/pytorch3d/tree/master/tests) for forward and backward computations for all the rendering and rasterization components in case they're helpful in your quest.

**Comment by achao2013** [comment]
the code like this:
outtex = network_texture(input)
outpts=network_shape(input)
veruvs= precomputed fixed values
tris = precomputed fixed values
tex = Textures(verts_uvs=veruvs, faces_uvs=tris, maps=outtex)
mesh = Meshes(verts=outpts, faces=tris, textures=tex)
R_, T_ = look_at_view_transform(2.7, 0, 0, device=device)
camas = OpenGLOrthographicCameras(device=device, R=R_, T=T_)
rasterizers=MeshRasterizer(
                    cameras=camas, 
                    raster_settings=raster_settings)
rendereras = MeshRenderer(
                        rasterizer=rasterizers,
                        shader=TexturedSoftPhongShader(
                            device=device, cameras=camas))
imageas = rendereras(mesh)
loss = l1loss(imageas , gtimg)
@gkioxari

**Comment by gkioxari** [comment]
@achao2013 You are giving us a sequence of commands which we are unable to do anything with. It's impossible to reproduce any of the things you pasted above. You need to break your code into parts, and run the different parts to figure out where your problem is. In this quest, you will be able to isolate the problem and find your bug. If you have strong evidence that the bug is in some PyTorch3D operation then you need to isolate it the operation and provide us code that will allow us to run it and reproduce it. So far, I have no evidence that the mistake lies in PyTorch3D and not some other part of your code and we can't go about trying to do magic to find where it is. So I am closing this issue for now and please re-open it once you have evidence the bug is in PyTorch3D and once you are able to give us reproducible code.
